### PR TITLE
Refactor Deletion Handling (untested)

### PR DESCRIPTION
**Pull request by LukvonStrom**
Code filters for null Elements and uses #forEach instead of #map and is more memory efficient

### CODE REVIEW

**Comment by LukvonStrom** [comment]
> This does not work for me, even with my try to fix syntax issues.
> 
> `const removeProperly = (...arrays) => { arrays.forEach(array => array.filter(n => !!n) .forEach(n => n.remove()) ); };`

It should work, could you check if the forEach method logs the correct values?

**Comment by tobimori** [comment]
`TypeError: array.forEach is not a function`

**Comment by LukvonStrom** [comment]
> `TypeError: array.forEach is not a function`
> 
> Apparently forEach does not work for a HTMLCollection.

does ist work now? I totally forgot about the whack implemented HTMLCollection :/

File: unpaywall-funke.user.js [hunk 1]
```diff
@@ -18,12 +18,23 @@
...
+const whitelist = [];
+
+const removeHTMLProperly = (...arrays) => {
+    [...arrays].forEach(([...array]) => array.filter(n => !!n)
+                                 .forEach(n => n.remove())
+                  );
+}
+
+const removeScriptsBalanced = (array) => {
+    removeHTMLProperly([...array].filter({src} => whitelist.filter(s => s.includes(src)).length > 0  ));
```
Comment (tobimori) [inline unpaywall-funke.user.js]: Remove the the curly brackets and it should work.

**Comment by LukvonStrom** [comment]
Nope, the curly brackets were intentional (=> object destructuring), I think normal brackets around the curly brackets should work

 

Von: Tobias <notifications@github.com> 
Gesendet: Montag, 9. März 2020 14:56
An: tobimori/unpaywall-funke <unpaywall-funke@noreply.github.com>
Cc: Lukas Fruntke <lukas@fruntke.tech>; Author <author@noreply.github.com>
Betreff: Re: [tobimori/unpaywall-funke] Refactor Deletion Handling (untested) (#5)

 

@tobimori requested changes on this pull request.

  _____  

In unpaywall-funke.user.js <https://github.com/tobimori/unpaywall-funke/pull/5#discussion_r389695284> :

> @@ -18,12 +18,23 @@
 // @run-at       document-start
 // ==/UserScript==
 
+const whitelist = [];
+
+const removeHTMLProperly = (...arrays) => {
+    [...arrays].forEach(([...array]) => array.filter(n => !!n)
+                                 .forEach(n => n.remove())
+                  );
+}
+
+const removeScriptsBalanced = (array) => {
+    removeHTMLProperly([...array].filter({src} => whitelist.filter(s => s.includes(src)).length > 0  ));

Remove the the curly brackets and it should work.

**Pull request merged**